Content.Server/Codewords/CodewordRuleComponent.cs

```diff
@@ -0,0 +1,23 @@
+﻿using Robust.Shared.Prototypes;
+
+namespace Content.Server.Codewords;
+
+/// <summary>
+/// Component that defines <see cref="CodewordGeneratorPrototype"/> to use and keeps track of generated codewords.
+/// </summary>
+[RegisterComponent, Access(typeof(CodewordSystem))]
+public sealed partial class CodewordRuleComponent : Component
```

Comment:
does it make sense to attach these codewords to rules? Like, if we have two syndicate rules with codewords, wouldn't we want them to share the codewords so long as they have the same faction prototype?

maybe it'd be better to store them on some arbitrary ent in nullspace like `ClockSystem`

Reply:
They do share them. If the game rule isn't added, it gets added when traitor calls for it. If its added twice (or more), it checks for first match, meaning you can have multiple codeword rules with different factions in them. 

I made them a gamerule because that allows easy access to a YAML config via the rule component, and it made the most sense in my head.